### ISSUE TITLE
TypeError: path.resolve is not a function.

### ISSUE THREAD

**Issue by Progressandro** [isaachinman/next-i18next#753]
Hey, thanks for coming up with an equally straightforward solution for the serverless option as well.

I'm trying to implement the update into my new project, however, I'm finding the following issue when importing the i18n file into any page-level component: `Error was not caught TypeError: path.resolve is not a function`

I'm not a veteran with Next.js, but I don't see any obvious problem with my code.

my i18n.js:
```javascript
const path = require('path')
const NextI18Next = require('next-i18next').default
const { localeSubpaths } = require('next/config').default().publicRuntimeConfig
module.exports = new NextI18Next({
  defaultLanguage: 'en',
  otherLanguages: ['es'],
  localeSubpaths,
  localePath: path.resolve('./public/locales'),
})
```

My "/test" page-level component:
```jsx
import React from 'react'
import PropTypes from 'prop-types'
import { i18n, withTranslation } from '../i18n'
const Test = ({ t }) => {
  return (
    <React.Fragment>
      <main>
        <div>
          <button
            type="button"
            onClick={() => {
              console.log(i18n)
              i18n.changeLanguage(i18n.language === 'en' ? 'es' : 'en')
            }}
          >
            Change
          </button>
          <h1>{t('title')}</h1>
        </div>
      </main>
    </React.Fragment>
  )
}

Test.getInitialProps = async () => ({
  namespacesRequired: ['common'],
})

Test.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation('common')(Test)
```
My next.config.js:
```javascript
const withPlugins = require('next-compose-plugins');
const withOptimizedImages = require('next-optimized-images');
const withFonts = require('next-fonts');
const FilterWarningsPlugin = require('webpack-filter-warnings-plugin');
const { nextI18NextRewrites } = require('next-i18next/rewrites')

const localeSubpaths = {
  es: 'es'
}

const nextConfig = {
  env: {
    GOOGLE_API_KEY: 'API_KEY',
    REACT_APP_GOOGLE_MAP_API_KEY:
      'https://maps.googleapis.com/maps/api/js?v=3.exp&key=API_KEY&libraries=geometry,drawing,places',
    SERVER_API: `http://localhost:3001`,
  },
  webpack: (config, { isServer }) => {
    config.plugins.push(
      new FilterWarningsPlugin({
        exclude: /mini-css-extract-plugin[^]*Conflicting order between:/,
      })
    );
    config.resolve.modules.push(__dirname);

    return config;
  },
  publicRuntimeConfig: {
    localeSubpaths,
  },
  experimental: {
    async rewrites() {
      return [
        ...nextI18NextRewrites(localeSubpaths)
      ]
    }
  }
};

module.exports = withPlugins(
  [
    [
      withOptimizedImages,
      {
        mozjpeg: {
          quality: 90,
        },
        webp: {
          preset: 'default',
          quality: 90,
        },
      },
    ],
    withFonts,
  ],
  nextConfig
);
```
If you have any other question, I'll be glad to provide more information if possible.

Thanks in advance! :muscle:

**Comment by isaachinman** [comment]
If you remove your webpack overrides, does the issue cease?

**Comment by Progressandro** [comment]
> If you remove your webpack overrides, does the issue cease?

No, it doesn't.

**Comment by isaachinman** [comment]
@Progressandro Mind putting together a reproducible repo? I'll debug from there. Thanks.

**Comment by Progressandro** [comment]
Already invited you to the repo, please use the next-i18next branch. Also, the route for testing it `/test`. Just try to go there and check the console error. Besides, if you try to click the "change" button, it won't work.

**Comment by isaachinman** [comment]
I believe it's something to do with [this line](https://github.com/Progressandro/homenet/blob/6d2315b74e14288177ebb7f3df9a8b7e39d0ab81/next.config.js#L26):

```jsx
config.resolve.modules.push(__dirname);
```

But cannot be sure, as this definitely isn't a minimal repro. Can you please remove all unnecessary code to demonstrate the potential bug in its simplest form?

Resolve comes [directly from webpack](https://webpack.js.org/configuration/resolve/), so I'm fairly confident this is user error.

**Comment by Progressandro** [comment]
I greatly apologize, Isaac. I know the example wasn't the best but I really wanted to give you a full picture since I didn't have an idea of where the problem was (this is legacy code for me).

Anyways, before almost giving up, for some reason I also reviewed the package.json, and I FOUND THE ISSUE:

```json
"browser": {
    "fs": false,
    "path": false
},
```

Removed `"path": false` and it works wonderfully now.

Thank you very much.
